design-document.md

```diff
@@ -1,8 +1,89 @@
 # Prio v3 Design Document
 
-## Sample user stories
+## Architecture overview
 
-## Threat model
+Prio is a system and protocol for privately computing aggregation functions over private 
+input. An aggregation function F is one that computes an output y = F(x1,x2,...) for inputs
+xi. In general, Prio supports any aggregation function whose inputs can be encoded in a 
+particular way. However, not all aggregation functions admit an efficient encoding, rendering
+them impractical to implement. Thus, Prio supports a limited set of aggregation functions, 
+some of which we highlight below:
+
+- Simple statistics, including sum, mean, min, max, variance, and standard deviation;
+- Bit vector OR and AND operations; and
+- Count-min sketch (approximated frequency counts) over a closed universe of strings.
+
+The applications for such aggregations functions are large, including, though not limited to:
+counting the number of times a sensitive or private event occurs and approximating the frequency
+that sensitive tokens or strings occur.
+
+Client applications hold private inputs to the aggregation function, server processors,
+or aggregators, invoke  multi-party computation to compute the output, and a final collector 
+obtains the output of the aggregation function. At a high level, the flow of data through
+these entities works roughly as follows:
+
+~~~
+                            +------------+     
+ (1) Batch submission       |            |        (3) Collection
+    +-----------------------> Aggregator +------------------+
+    |                       |            |                  |
+    |                       +-^-------^--+                  |
+    |                         |       |                     |
+    |                         |       |                     |
+    |                         |       |  (2) MPC            |
++--------+           +--------v---+   |      eval      +----v------+
+|        |           |            |   |                |           |
+| Client +-----------> Aggregator |   |                | Collector |
+|        |           |            |   |                |           |
++--------+           +--------^---+   |                +----^------+
+    |                         |       |                     |
+    |                         |       |                     | 
+    |                         |       |                     |
+    |                       +-v-------v--+                  |
+    |                       |            |                  |
+    +-----------------------> Aggregator +------------------+
+                            |            |
+                            +------------+
```

Comment:
Pretty! Though I think we need to align this with the candidate protocol #2. In particular, I would remove the wires going from the client to the top and bottom aggregators, and I would rename the middle aggregator to "Leader'.